src/ert/gui/model/job_list.py

```diff
@@ -129,65 +126,51 @@ def parent(self, child: Optional[QModelIndex] = None) -> Optional[QObject]:
     def index(
         self, row: int, column: int, parent: Optional[QModelIndex] = None
     ) -> QModelIndex:
-        if parent is None:
-            parent = QModelIndex()
-        if parent.isValid():
-            return QModelIndex()
-        job_index = self.mapToSource(self.createIndex(row, column, parent))
-        return self.createIndex(row, column, job_index.data(NodeRole))
+        parent = parent if parent else QModelIndex()
+        if not parent.isValid():
+            job_index = self.mapToSource(self.createIndex(row, column, parent))
+            return self.createIndex(row, column, job_index.data(NodeRole))
+        return QModelIndex()
 
     def mapToSource(self, proxyIndex: QModelIndex) -> QModelIndex:
-        if not proxyIndex.isValid():
-            return QModelIndex()
-        source_model = self.sourceModel()
-        assert source_model is not None
-        iter_index = source_model.index(self._iter, 0, QModelIndex())
-        if not iter_index.isValid() or not source_model.hasChildren(iter_index):
-            return QModelIndex()
-        real_index = source_model.index(self._real, 0, iter_index)
-        if not real_index.isValid() or not source_model.hasChildren(real_index):
-            return QModelIndex()
-        job_index = source_model.index(
-            proxyIndex.row(), proxyIndex.column(), real_index
-        )
-        return job_index
+        if proxyIndex.isValid():
+            sm = self.sourceModel()
+            assert sm is not None
+            iter_index = sm.index(self._iter, 0, QModelIndex())
+            if iter_index.isValid() and sm.hasChildren(iter_index):
+                real_index = sm.index(self._real, 0, iter_index)
+                if real_index.isValid() and sm.hasChildren(real_index):
+                    return sm.index(proxyIndex.row(), proxyIndex.column(), real_index)
+        return QModelIndex()
 
-    def mapFromSource(self, sourceIndex: QModelIndex) -> QModelIndex:
-        if not sourceIndex.isValid():
-            return QModelIndex()
-        if not self._accept_index(sourceIndex):
-            return QModelIndex()
-        return self.index(sourceIndex.row(), sourceIndex.column(), QModelIndex())
+    def mapFromSource(self, src_index: QModelIndex) -> QModelIndex:
+        if src_index.isValid():
+            source_node = src_index.internalPointer()
+
+            if source_node and self._accept_index(src_index):
+                return self.index(src_index.row(), src_index.column(), QModelIndex())
+        return QModelIndex()
```

Comment:
Do we need the added internalPointer check?
It is also checked in _accept_index().
This could be simplified to
`return self.index(src_index.row(), src_index.column(), QModelIndex()) if src_index.isValid() and self._accept_index(src_index) else QModelIndex()`

Reply:
Nice! I have replaced this proposal. 👍